doc/syslog/Syslog_Source_IP_test_plan.md

```diff
@@ -0,0 +1,213 @@
+7. Repeat steps 1-6 by setting different port, and check the port of syslog message will be changed to the set one
+
+
+### Test cases #4 -  Configure syslog server with VRF/Source: set/set
+1. For vrf: default, mgmt, Vrf-data, Configure syslog server with VRF/Source: set/set like below:
+```
+config syslog add 2.2.2.2 ---source 1.1.1.1 --vrf default
+config syslog add 3.3.3.3 ---source 5.5.5.5 --vrf mgmt
+config syslog add 2222::2222 ---source 1111::1111 --vrf Vrf-data
+```
+2. Check syslog config by show syslog, the result should like below:
+```
+# show syslog
+SERVER      SOURCE      PORT    VRF
+----------  ----------  ------  --------
+2.2.2.2     1.1.1.1      514     default
+3.3.3.3     5.5.5.5      514     mgmt
+2222::2222  1111::1111   514     Vrf-data
+```
+3. Check the corresponding interface will send syslog message with port 514 and related source IP on dut
+4. Del syslog server config like below:
+```
+config syslog del 2.2.2.2
+config syslog del 3.3.3.3
+config syslog del 2222::2222 
+```
+5. Check syslog config by show syslog, the relevant config should be removed
+6. Check the corresponding interface will not send any syslog message with related port and source ip
+7. Repeat steps 1-6 by setting different port, and check the port of syslog message will be changed to the set one
+
+
+### Test cases #5 -  Disable mgmt VRF or remove data VRF exists in syslog config, there will be an error prompt
+1. Configure syslog sever with VRF/Source: set/set like below:
+```
+config syslog add 3.3.3.3 --source 5.5.5.5 --vrf mgmt
+config syslog add 2222::2222 --source 1111::1111 --vrf Vrf-data
+```
+2. Check syslog config by show syslog, the result should like below:
+```
+# show syslog
+SERVER      SOURCE      PORT    VRF
+----------  ----------  ------  --------
+3.3.3.3     5.5.5.5      514     mgmt
+2222::2222  1111::1111   514     Vrf-data
+```
+3. Disable mgmt VRF or remove Data VRF
+4. Check there is an error prompt such as: VRF exists in syslog config, it can not be disabled or removed.
+
+
+### Test cases #6 -  After Configure syslog server and save config, do cold/fast/warm reboot, check syslog config still work
+1. Configure syslog server with VRF/Source: set/set like below:
+```
+config syslog add 2.2.2.2 --source 1.1.1.1 --vrf default
+config syslog add 3.3.3.3 --source 5.5.5.5 --vrf mgmt
+config syslog add 2222::2222 --source 1111::1111 --vrf Vrf-data
+```
+2. Check syslog config by show syslog, the result should like below:
+```
+# show syslog
+SERVER      SOURCE      PORT    VRF
+----------  ----------  ------  --------
+2.2.2.2     1.1.1.1      514     default
+3.3.3.3     5.5.5.5      514     mgmt
+2222::2222  1111::1111   514     Vrf-data
+```
+3. Check the corresponding interface will send syslog message with port 514 and related source IP on dut
+4. Save config and do cold/fast/warm reboot
+5. Check Syslog config still work
+
+
+ ### Test cases #7 -  Configure syslog server with invalid source IP
```

Comment:
@JibinBao suggest to verify the case when IP is not configured on any VRF member

Reply:
yes. in the step4(non-existing IP ), we have it.
4. Repeat step 1~3 with non-existing IP and multicast IP